src/pocket/PocketPagerDuty.ts

```diff
@@ -0,0 +1,180 @@
+import { Construct } from 'constructs';
+import {
+  DataPagerdutyVendor,
+  Service,
+  ServiceIntegration,
+} from '../../.gen/providers/pagerduty';
+import { SnsTopic, SnsTopicSubscription } from '../../.gen/providers/aws';
+
+export interface PocketPagerDutyProps {
+  prefix: string;
+  service: {
+    autoResolveTimeout?: number;
+    acknowledgementTimeout?: number;
+    criticalEscalationPolicyId: string;
+    nonCriticalEscalationPolicyId: string;
+  };
+  sns?: {
+    topic?: {
+      tags?: { [key: string]: string };
+    };
+    subscription?: {
+      confirmationTimeoutInMinutes?: number;
+    };
+  };
+}
+
+export enum PAGERDUTY_SERVICE_URGENCY {
+  CRITICAL = 'Critical',
+  NON_CRITICAL = 'Non-Critical',
+}
+
+export class PocketPagerDuty extends Resource {
+  public readonly snsCriticalAlarmTopic: SnsTopic;
+  public readonly snsNonCriticalAlarmTopic: SnsTopic;
+  static readonly SERVICE_AUTO_RESOLVE_TIMEOUT = '14400';
+  static readonly SERVICE_ACKNOWLEDGEMENT_TIMEOUT = '600';
+  static readonly SNS_SUBSCRIPTION_CONFIRMATION_TIMEOUT_IN_MINUTES = 2;
+  private config: PocketPagerDutyProps;
+
+  constructor(scope: Construct, name: string, config: PocketPagerDutyProps) {
+    super(scope, name);
+
+    this.config = config;
+
+    const sentryVendor = this.getVendor('Sentry');
+
+    const cloudwatchVendor = this.getVendor('Cloudwatch');
+
+    const pagerDutyCritical = this.createService(
+      PAGERDUTY_SERVICE_URGENCY.CRITICAL
+    );
+
+    const pagerDutyNonCritical = this.createService(
+      PAGERDUTY_SERVICE_URGENCY.NON_CRITICAL
+    );
+
+    this.createServiceIntegration(
+      sentryVendor,
+      pagerDutyCritical,
+      PAGERDUTY_SERVICE_URGENCY.CRITICAL
+    );
+
+    this.createServiceIntegration(
+      sentryVendor,
+      pagerDutyNonCritical,
+      PAGERDUTY_SERVICE_URGENCY.NON_CRITICAL
+    );
+
+    const cloudwatchCriticalIntegration = this.createServiceIntegration(
+      cloudwatchVendor,
+      pagerDutyCritical,
+      PAGERDUTY_SERVICE_URGENCY.CRITICAL
+    );
+
+    const cloudwatchNonCriticalIntegration = this.createServiceIntegration(
+      cloudwatchVendor,
+      pagerDutyNonCritical,
+      PAGERDUTY_SERVICE_URGENCY.NON_CRITICAL
+    );
+
+    const snsCriticalAlarmTopic = (this.snsCriticalAlarmTopic = this.createSnsTopic(
+      PAGERDUTY_SERVICE_URGENCY.CRITICAL
+    ));
+
+    const snsNonCriticalAlarmTopic = (this.snsNonCriticalAlarmTopic = this.createSnsTopic(
+      PAGERDUTY_SERVICE_URGENCY.NON_CRITICAL
+    ));
+
+    this.createSnsTopicSubscription(
+      snsCriticalAlarmTopic,
+      cloudwatchCriticalIntegration,
+      PAGERDUTY_SERVICE_URGENCY.CRITICAL
+    );
+
+    this.createSnsTopicSubscription(
+      snsNonCriticalAlarmTopic,
+      cloudwatchNonCriticalIntegration,
+      PAGERDUTY_SERVICE_URGENCY.NON_CRITICAL
+    );
+  }
+
+  private createSnsTopicSubscription(
+    topic: SnsTopic,
+    integration: ServiceIntegration,
+    urgency: PAGERDUTY_SERVICE_URGENCY
+  ) {
```

Comment:
can we have return types on these functions?

Reply:
done